api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto

```diff
@@ -174,4 +175,8 @@ message HttpProtocolOptions {
   // [#not-implemented-hide:]
   // [#extension-category: envoy.http.header_validators]
   config.core.v3.TypedExtensionConfig header_validation_config = 7;
+
+  // If specified, only responses matching the matcher will be treated by outlier detection as errors.
+  // If not specified, only 5xx codes are treated by outlier detection as errors.
+  config.common.matcher.v3.MatchPredicate outlier_detection_error_matcher = 8;
```

Comment:
I actually prefer you previous design about add a new `OutlierDetection` message. We may could add new field at there in the future if we aware more feature requirements. (If I didn't present this clearly before, sorry! 😞 )

```
message OutlierDetection {
  config.common.matcher.v3.MatchPredicate failure_matcher = 1;
}
```